content/development/testing.md

```diff
@@ -9,7 +9,312 @@ chapter: false
+	running 913100-3: ✔ passed in 70.200211ms (RTT 76.92281ms)
+	running 913100-4: ✔ passed in 65.856005ms (RTT 73.328341ms)
+	running 913100-5: ✔ passed in 66.986859ms (RTT 73.494356ms)
+  ...
+```
+
+To run the test suite against nginx, execute the following:
+
+```bash
+$ ./go-ftw run --config .ftw.nginx.yaml -d ../coreruleset/tests/regression/tests/
+🛠️  Starting tests!
+🚀 Running go-ftw!
+👉 executing tests in file 911100.yaml
+	running 911100-1: ✔ passed in 851.460335ms (RTT 292.802335ms)
+	running 911100-2: ✔ passed in 53.748811ms (RTT 66.798867ms)
+	running 911100-3: ✔ passed in 49.237535ms (RTT 67.964411ms)
+	running 911100-4: ✔ passed in 194.935023ms (RTT 202.414171ms)
+	running 911100-5: ✔ passed in 52.905305ms (RTT 66.254034ms)
+	running 911100-6: ✔ passed in 52.597784ms (RTT 68.58854ms)
+	running 911100-7: ✔ passed in 51.996881ms (RTT 67.496534ms)
+	running 911100-8: ✔ passed in 50.804143ms (RTT 67.589557ms)
+👉 executing tests in file 913100.yaml
+	running 913100-1: ✔ passed in 276.383507ms (RTT 85.436758ms)
+	running 913100-2: ✔ passed in 86.682684ms (RTT 69.89541ms)
+  ...
+```
+
+If you want to run only one test, or a group of tests, you can specify that using the "include" option `-i` (or `--include`). This option takes a regular expression:
+
+```bash
+$ ./go-ftw run --config .ftw.apache.yaml -d ../coreruleset/tests/regression/tests/ -i "955100-1$"
+```
+
+In the above case only the test case `955100-1` will be run.
+
+If you need to see more verbose output (e.g., to look at the requests and responses sent and received by **go-ftw**) you can use the `--debug` or `--trace` options:
+
+```bash
+$ ./go-ftw run --config .ftw.apache.yaml -d ../coreruleset/tests/regression/tests/ -i "955100-1$" --trace
+```
+
+
+```bash
+$ ./go-ftw run --config .ftw.apache.yaml -d ../coreruleset/tests/regression/tests/ -i "955100-1$" --debug
+```
+
+Please note again that `libmodsecurity3` is **not fully compatible** with ModSecurity 2, some tests can fail. If you want to ignore them, you can put the tests into a list in your config:
+
+```
+testoverride:
+  input:
+    dest_addr: "127.0.0.1"
+    port: 8080
+  ignore:
+    # text comes from our friends at https://github.com/digitalwave/ftwrunner
+    '941190-3$': 'known MSC bug - PR #2023 (Cookie without value)'
+    '941330-1$': 'know MSC bug - #2148 (double escape)'
+    ...
+```
+
+For more information and examples, please check the [go-ftw documentation](https://github.com/coreruleset/go-ftw#example-usage).
+
+**Also please don't forget to roll back the modifications from this guide to your WAF configuration after you're done testing!**
+
+## Ftw
+
+[ftw](https://github.com/coreruleset/ftw) is our legacy test runner, retired with CRS 4.0. We used to automatically run our test suite using **ftw** as part of a [GitHub workflow](https://github.com/coreruleset/coreruleset/blob/v3.3/dev/.github/workflows/test.yml). You can easily reproduce that locally, on your workstation.
```

Comment:
```suggestion
[ftw](https://github.com/coreruleset/ftw) is our legacy test runner, and is **deprecated** and no longer used for tests after CRS v3.2. Refer to this [GitHub workflow](https://github.com/coreruleset/coreruleset/blob/v3.3/dev/.github/workflows/test.yml) if you need to use the old **ftw** Python version. You can easily reproduce that locally, on your workstation.
```

Reply:
We need to change the link too, it refers to v3.3 workflow (where we use go-ftw)

Reply:
Done